crates/apollo-compiler/tests/input_coercion.rs

```diff
@@ -0,0 +1,50 @@
+
+#[test]
+fn test_graphql_float_variable_coercion() {
+    // Small schema with a Float in the input object
+    let sdl = r#"
+      type Car { id: ID! kilometers: Float! }
+      input CarInput { kilometers: Float! }
+      type Query { getCarById(id: ID!): Car }
+      type Mutation { insertACar(car: CarInput!): Car!
+      }
```

Comment:
Nit: this might be a bit easier to read.
```suggestion
      type Mutation { insertACar(car: CarInput!): Car! }
```